Source/Timeline/CurrentTimeIndicator.swift

```diff
@@ -36,14 +39,32 @@ class CurrentTimeIndicator: UIView {
     [timeLabel, circle, line].forEach {
       addSubview($0)
     }
+    
+    //Allow label to adjust so that am/pm can be displayed if format is changed.
+    timeLabel.numberOfLines = 1
+    timeLabel.adjustsFontSizeToFitWidth = true
+    timeLabel.minimumScaleFactor = 0.5
+    
+    //The width of the label is determined by leftInset and padding. 
+    //The y position is determined by the line's middle.
+    timeLabel.translatesAutoresizingMaskIntoConstraints = false
+    timeLabel.widthAnchor.constraint(equalToConstant: leftInset - (3 * padding)).isActive = true
+    timeLabel.rightAnchor.constraint(equalTo: line.leftAnchor, constant: -padding).isActive = true
+    timeLabel.centerYAnchor.constraint(equalTo: line.centerYAnchor).isActive = true
+    timeLabel.baselineAdjustment = .alignCenters
+    
     updateStyle(style)
   }
+    
+  func updateDate() {
+    let dateFormat = is24hClock ? "HH:mm" : "h:mm a"
+    timeLabel.text = date.format(with: dateFormat)
+    timeLabel.sizeToFit()
+    setNeedsLayout()
+  }
 
   override func layoutSubviews() {
-    let size = timeLabel.frame.size
-    timeLabel.align(Align.toTheLeftCentered, relativeTo: line, padding: 5, width: size.width, height: size.height)
```

Comment:
I had some trouble getting the height of the label correct in the case that the string was too long to fit. I used constraints for layout instead (line 50-54)